include/openthread/netdiag.h

```diff
@@ -202,6 +202,14 @@ typedef struct otNetworkDiagChildEntry
      */
     uint16_t mTimeout : 5;
 
+    /**
+     * Link Quality In value in [0,3].
+     *
+     * Value 0 indicates that sender does not support the feature to provide link quality info.
+     *
+     */
+    uint8_t mLinkQuality : 2;
+
```

Comment:
It is strongly suggest that we should avoid using bitfield, because bit endianness is implementation defined. (Do not confuse with byte endianness, which is a totally different thing).

I saw this class `otNetworkDiagChildEntry` is only used internally in diagnostic, which may be fine now, but I suggest move to offset+mask, as same as we do in `ChildTableEntry` class.

Reply:
> because bitfield endianness is implementation defined

Yes. That's well-known. It is not allowed in OT core for encoding data which is appended/parsed in messages for this reason.  It can be only used for storing values in more compact format.

This is public API/definition and intended to pass the values back to user.